sympy/ntheory/factor_.py

```diff
@@ -973,7 +974,7 @@ def done(n, d):
 
 
 def factorint(n, limit=None, use_trial=True, use_rho=True, use_pm1=True,
-              verbose=False, visual=None, multiple=False):
+              use_ecm = False, verbose=False, visual=None, multiple=False):
```

Comment:
The `use_xxx` flags are probably not much needed by ordinary users. How about setting the default  `use_ecm=True` (with no spaces around =) and then changing it internally to `False` for small `n` to save time?

Reply:
Done. Now my idea is that if ecm=true, then perform pollard rho method for 3 iterations then if after this the num of digits is >=20 then perform ecm.

Reply:
@jksuom for some reason, 1 test of test_matrices are failing.